com.unity.ml-agents/Tests/Editor/MLAgentsEditModeTest.cs

```diff
@@ -602,4 +603,56 @@ public void TestHeuristicPolicyStepsSensors()
             Assert.AreEqual(numSteps, agent1.sensor2.numCompressedCalls);
         }
     }
+
+    [TestFixture]
+    public class TestOnEnableOverride
+    {
+        public class OnEnableAgent : Agent
+        {
+            public bool callBase;
+
+            protected override void OnEnable()
+            {
+                if (callBase)
+                    base.OnEnable();
+            }
+        }
+
+        [Test]
+        public void TestAgentDontCallBaseOnEnable()
+        {
+            var go = new GameObject();
+            var agent = go.AddComponent<OnEnableAgent>();
+            var onEnable = typeof(OnEnableAgent).GetMethod("OnEnable", BindingFlags.NonPublic | BindingFlags.Instance);
+            var sendInfo = typeof(Agent).GetMethod("SendInfoToBrain", BindingFlags.NonPublic | BindingFlags.Instance);
+            Assert.NotNull(onEnable);
+            onEnable.Invoke(agent, null);
+            Assert.NotNull(sendInfo);
+            Assert.Throws<UnityAgentsException>(() =>
```

Comment:
I'm confused, why does this result in a `UnityAgentsException`?

Is it worth combining these to make it clearer that the setup is identical (and make it less likely they diverge in the future)?

Something like
```c#
void TestLifeCycleMethods(bool callbase)
{
  // do the setup
  if(!callbase) {
    Assert.Throws<..>...
  } else {
    Assert.DoesNotThrow...
  }
}

[Test]
public void TestAgentDontCallBaseOnEnable()
{
  TestLifeCycleMethods(false);
}

[Test]
public void TestAgentCallBaseOnEnable()
{
  TestLifeCycleMethods(true);
}

```

Reply:
The Reflection Target Invoker catches the real exception, then throws its own.  So I catch the TargetInvocationException and throw the base exception (which is the one I'm expecting). Sure I can combine them.

Reply:
But what's the real one? I didn't see any `UnityAgentsException` thrown from `SendInfoToBrain()`

Reply:
Doh, I was looking locally, missed the new exception.